### ISSUE TITLE
Loss is exploding, aborting

### ISSUE THREAD

**Issue by mehanelson** [karpathy/char-rnn#169]
I am using an amazon ec2 instance ubuntu 14.04 with gpu (g2.8xlarge). I am running the code on an input.txt file of size 221 MB. The code after running for 4 days, gave the following error:
Loss is exploding, aborting

Can you please give some ideas/suggestions on how to fix this problem? What should one do to run on large input files. Any thoughts on this are welcome.
Thanks!

[I tested the code on smaller input files, and it works perfectly fine]

**Comment by soumith** [comment]
try lowering the learning rate.

**Comment by lizhihuan** [comment]
I comment  the statement at the end of the code file `if loss[1] > loss0 * 3 then`, and it can run successfully. The train loss descend as i expected.

**Comment by mehanelson** [comment]
Thank you! I shall try both and get back to you once I get the results.

**Comment by mehanelson** [comment]
@lizhihuan I tried commenting it and it no longer gives me the error. Thank you!

**Comment by mehanelson** [comment]
I am running on a huge dataset. I started running on July 5th and it still running and may run for a couple of weeks. The loss had reached from 0.9 to 17 suddenly so I was getting the error. However, on commenting the statement mentioned by @lizhihuan  the program didn't abort when the same thing happened. The loss reduced considerably to 3 after reaching 17 but its still a high value. Last time, I checked it had reached 7. So, lowering the learning rate help or is there something else I need to do?
Thanks!
